### PR TITLE
[node]: Use specific UUID pattern return type for `webcrypto.Crypto.randomUUID`

### PR DESCRIPTION
**Pull request by laymonage**
Fixes #64798.

It's outside of the original scope, but I also updated the types for `crypto.randomUUID` as I think it can be useful to be consistent for both.

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: #64798

### CODE REVIEW

**Comment by Semigradsky** [comment]
Looks like in `ts<=4.8` version we can remain `string`.

**Comment by laymonage** [comment]
Good call, I've amended the commit to not apply the changes to the `ts4.8` directories. Thanks!

File: types/node/crypto.d.ts [hunk 1]
```diff
@@ -3134,7 +3134,7 @@ declare module 'crypto' {
      * cryptographic pseudorandom number generator.
      * @since v15.6.0, v14.17.0
      */
-    function randomUUID(options?: RandomUUIDOptions): string;
+    function randomUUID(options?: RandomUUIDOptions): `${string}-${string}-${string}-${string}-${string}`;
```
Comment (SimonSchick) [inline types/node/crypto.d.ts:3137]: Can we wrap this into a `UUID` type?

**Review by Semigradsky — APPROVED** [review]

**Review by peterblazejewicz — APPROVED** [review]
LGTM!
@laymonage thanks!

**Comment by laymonage** [comment]
Ready to merge

**Pull request merged**